### ISSUE TITLE
FTS tables cause broken .sql export

### ISSUE THREAD

**Issue by simonw** [simonw/datasette-backup#5]
Found while building https://github.com/simonw/datasette-backup/issues/2#issuecomment-687717120

If your database contains a FTS table, the dumped SQL from `/-/backup/dbname.sql` fails to create a database when piped into `sqlite3`:

```
~ % cat /tmp/broken.sql | sqlite3 /tmp/dogs.db
Error: near line 15: table 'dogs_fts_config' already exists
Error: near line 16: UNIQUE constraint failed: dogs_fts_config.k
Error: near line 17: table 'dogs_fts_data' already exists
Error: near line 18: UNIQUE constraint failed: dogs_fts_data.id
Error: near line 19: UNIQUE constraint failed: dogs_fts_data.id
Error: near line 21: table 'dogs_fts_docsize' already exists
Error: near line 22: UNIQUE constraint failed: dogs_fts_docsize.id
Error: near line 23: UNIQUE constraint failed: dogs_fts_docsize.id
Error: near line 24: table 'dogs_fts_idx' already exists
Error: near line 26: constraint failed
```

**Comment by simonw** [comment]
Here's the SQL that came back from `/-/backup/fts.sql`:
```sql
BEGIN TRANSACTION;
CREATE TABLE [dogs] (
   [id] INTEGER PRIMARY KEY,
   [name] TEXT,
   [age] INTEGER
);
INSERT INTO "dogs" VALUES(1,'Cleo',5);
INSERT INTO "dogs" VALUES(2,'Pancakes',4);
CREATE VIRTUAL TABLE [dogs_fts] USING FTS5 (
                [name],
                content=[dogs]
            );
INSERT INTO "dogs_fts" VALUES('Cleo');
INSERT INTO "dogs_fts" VALUES('Pancakes');
CREATE TABLE 'dogs_fts_config'(k PRIMARY KEY, v) WITHOUT ROWID;
INSERT INTO "dogs_fts_config" VALUES('version',4);
CREATE TABLE 'dogs_fts_data'(id INTEGER PRIMARY KEY, block BLOB);
INSERT INTO "dogs_fts_data" VALUES(1,X'0202');
INSERT INTO "dogs_fts_data" VALUES(10,X'000000000101010001010101');
INSERT INTO "dogs_fts_data" VALUES(137438953473,X'0000001A0530636C656F010202010870616E63616B65730202020409');
CREATE TABLE 'dogs_fts_docsize'(id INTEGER PRIMARY KEY, sz BLOB);
INSERT INTO "dogs_fts_docsize" VALUES(1,X'01');
INSERT INTO "dogs_fts_docsize" VALUES(2,X'01');
CREATE TABLE 'dogs_fts_idx'(segid, term, pgno, PRIMARY KEY(segid, term)) WITHOUT ROWID;
INSERT INTO "dogs_fts_idx" VALUES(1,X'',2);
COMMIT;
```
This appears to be a bug in Python's implementation of `iterdump()`. On closer inspection that connection method calls code in the `sqlite3/dump.py` module that ships with Python: https://github.com/python/cpython/blob/v3.8.5/Lib/sqlite3/dump.py

This appears to be a pretty simple emulation of how `.dump` works in SQLite itself. The key clue is the commented out code here:

```python
    for table_name, type, sql in schema_res.fetchall():
        if table_name == 'sqlite_sequence':
            yield('DELETE FROM "sqlite_sequence";')
        elif table_name == 'sqlite_stat1':
            yield('ANALYZE "sqlite_master";')
        elif table_name.startswith('sqlite_'):
            continue
        # NOTE: Virtual table support not implemented
        #elif sql.startswith('CREATE VIRTUAL TABLE'):
        #    qtable = table_name.replace("'", "''")
        #    yield("INSERT INTO sqlite_master(type,name,tbl_name,rootpage,sql)"\
        #        "VALUES('table','{0}','{0}',0,'{1}');".format(
        #        qtable,
        #        sql.replace("''")))
        else:
            yield('{0};'.format(sql))
```
So it looks like `conn.iterdump()` does not yet correctly handle virtual tables.

**Comment by simonw** [comment]
This code looks to mostly be 13 years old! https://github.com/python/cpython/commit/b1b9382d91e4b2e863225179cde4a61f0300a233

**Comment by simonw** [comment]
So... I may have to write my own, updated version of `dump.py` to get this to work with common database shapes used with Datasette!

**Comment by simonw** [comment]
Surprisingly the code in `dump.py` is simple enough that I actually think it would be practical to ship my own fixed implementation with this plugin.

**Comment by simonw** [comment]
If I do this I'm going to build it as a separate Python library with its own test suite.

**Comment by simonw** [comment]
I've released [sqlite-dump 0.1](https://github.com/simonw/sqlite-dump/releases/tag/0.1) which should help me fix this.
